sparkctl/cmd/gcs.go

```diff
@@ -18,111 +18,62 @@ package cmd
 
 import (
 	"fmt"
-	"io"
-	"os"
-	"path/filepath"
 
 	"cloud.google.com/go/storage"
+	"github.com/google/go-cloud/blob/gcsblob"
+	"github.com/google/go-cloud/gcp"
 	"golang.org/x/net/context"
 )
 
-const rootPath = "spark-app-dependencies"
-
-type gcsUploader struct {
-	client *storage.Client
-	handle *storage.BucketHandle
-	bucket string
-	path   string
+type GCSBlob struct {
+	projectId string
+	endpoint  string
+	region    string
 }
 
-func newGcsUploader(bucket string, path string, projectID string, ctx context.Context) (*gcsUploader, error) {
+func (blob GCSBlob) setPublicACL(
+	ctx context.Context,
+	bucket string,
+	filePath string) error {
 	client, err := storage.NewClient(ctx)
```

Comment:
So we still need to use this gcs-specific client for setting public ACL?

Reply:
Yes. This already kills a lot of redundant code, but ACL are not implemented in  `cloud-go` yet. This `cloud-go` seems a very young project.

Reply:
OK, can we add a `TODO` to get rid of this gcs-specific code once `cloud-go` adds support for that?

Reply:
Done